src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternFactory.cs

```diff
@@ -357,6 +373,14 @@ RoutePatternPart VisitPart(RoutePatternPart part)
                     parameterConstraints.AddRange(parameter.Constraints);
                 }
 
+                if (Equals(parameter.Default, @default)
+                    && parameter.Constraints.Count == 0
+                    && (parameterConstraints?.Count ?? 0) == 0)
```

Comment:
Is this ok? I choose to be conservative and force a copy if there are any constraints.